IronFoundry.Container.Shared/ExtensionMethods.cs

```diff
@@ -81,8 +82,32 @@ public static class ActionExtensionMethods
             return threwException;
         }
     }
+
+    public static class FuncExtensionMethods
+    {
+        /// <summary>
+        /// Retries the specified action until it returns true or up to the specified number
+        /// of retries.
+        /// </summary>
+        public static void RetryUpToNTimes(this Func<bool> action, int maxRetry, int sleepInMilli = 200)
```

Comment:
We should consider moving to some form of injectable retry strategy so we can control / verify it during tests.